packages/store/addon/-private/identifiers/cache.ts

```diff
@@ -340,17 +340,15 @@ export class IdentifierCache {
   ): StableRecordIdentifier {
     let identifier = this.getOrCreateRecordIdentifier(identifierObject);
 
-    let id = identifier.id;
     let newId = coerceId(data.id);
-
-    const keyOptions = getTypeIndex(this._cache.types, identifier.type);
     let existingIdentifier = detectMerge(this._cache.types, identifier, data, newId, this._cache.lids);
 
     if (existingIdentifier) {
+      const keyOptions = getTypeIndex(this._cache.types, identifier.type);
```

Comment:
While you are cleaning up, it would be great to not have this const here, as we do a let keyOptions on line 357